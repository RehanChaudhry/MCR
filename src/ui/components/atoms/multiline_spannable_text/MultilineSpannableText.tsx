import React from "react";
import { StyleProp, Text, View, ViewStyle } from "react-native";
import {
  AppLabel,
  AppLabelProps
} from "ui/components/atoms/app_label/AppLabel";

export interface SpannableProps {
  containerStyle?: StyleProp<ViewStyle>;
  text: Array<string>;
  appLabelProps: Array<AppLabelProps>;
}

type Props = SpannableProps;

const MultilineSpannableText = React.memo<Props>(
  ({ text, containerStyle, appLabelProps }) => {
    if (
      text.length > 0 &&
      appLabelProps.length > 0 &&
      text.length === appLabelProps.length
    ) {
      return (
        <View testID={"SPANNABLE_TEXT"} style={containerStyle}>
          <Text>
            {text.map((item, index) => (
              <AppLabel
                key={index}
                text={item}
                {...appLabelProps[index]}
              />
            ))}
          </Text>
        </View>
      );
    } else {
      return <View />;
    }
  }
);

export default MultilineSpannableText;
