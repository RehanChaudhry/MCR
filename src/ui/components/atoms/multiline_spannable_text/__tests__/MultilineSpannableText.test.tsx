import { fireEvent, render } from "@testing-library/react-native";
import React from "react";
import MultilineSpannableText from "ui/components/atoms/multiline_spannable_text/MultilineSpannableText";
import { StyleSheet } from "react-native";

test("check render properly", () => {
  const { getByTestId } = render(
    <MultilineSpannableText
      text={[
        { id: 1, text: "Male" },
        { id: 2, text: "Female" },
        { id: 3, text: "Others" }
      ]}
      textStyle={[styles.firstText, styles.email, styles.thirdText]}
    />
  );
  const spannableText = getByTestId("SPANNABLE_TEXT");
  expect(spannableText).toBeDefined();
});

test("Snapshot testing", () => {
  const rendered = render(
    <MultilineSpannableText
      text={[
        { id: 1, text: "Male" },
        { id: 2, text: "Female" },
        { id: 3, text: "Others" }
      ]}
      textStyle={[styles.firstText, styles.email, styles.thirdText]}
    />
  ).toJSON();
  expect(rendered).toMatchSnapshot();
});

test("check spannable text on Press", () => {
  const onPress = jest.fn();
  const { getByText } = render(
    <MultilineSpannableText
      text={[
        { id: 1, text: "Male" },
        { id: 2, text: "Female" },
        { id: 3, text: "Others" }
      ]}
      textStyle={[styles.firstText, styles.email, styles.thirdText]}
      onPress={onPress}
    />
  );
  const spannableText = getByText("Male");
  fireEvent.press(spannableText);
  expect(onPress).toBeCalledTimes(1);
});

const styles = StyleSheet.create({
  firstText: {
    fontWeight: "normal"
  },
  email: {
    fontWeight: "bold"
  },
  thirdText: {
    fontWeight: "normal"
  }
});
