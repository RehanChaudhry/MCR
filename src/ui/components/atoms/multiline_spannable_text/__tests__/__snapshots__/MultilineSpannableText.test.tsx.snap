// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing 1`] = `
<View
  testID="SPANNABLE_TEXT"
>
  <Text>
    <Text
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "fontWeight": "normal",
          },
        ]
      }
    >
      Male
    </Text>
    <Text
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "fontWeight": "bold",
          },
        ]
      }
    >
      Female
    </Text>
    <Text
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "fontWeight": "normal",
          },
        ]
      }
    >
      Others
    </Text>
  </Text>
</View>
`;
